### ISSUE TITLE
cuda tensor will raise an exception in np.asanyarray

### ISSUE THREAD

**Issue by Trinkle23897** [thu-ml/tianshou#152]
```python
In [2]: a=Batch(a={})

In [3]: b=Batch(a=torch.tensor([3,4], device='cuda'))

In [4]: Batch.cat([a,b])
```
will raise an exception. Same as Batch.stack:
```python
~/github/tianshou-new/tianshou/data/batch.py in _is_scalar(value)
     37     # the check of dict / Batch is omitted because this only checks a value.
     38     # a dict / Batch will eventually check their values
---> 39     value = np.asanyarray(value)
     40     return value.size == 1 and not value.shape
     41 

~/.local/lib/python3.6/site-packages/numpy/core/_asarray.py in asanyarray(a, dtype, order)
    136 
    137     """
--> 138     return array(a, dtype, copy=False, order=order, subok=True)
    139 
    140 

~/.local/lib/python3.6/site-packages/torch/tensor.py in __array__(self, dtype)
    490     def __array__(self, dtype=None):
    491         if dtype is None:
--> 492             return self.numpy()
    493         else:
    494             return self.numpy().astype(dtype, copy=False)

TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

**Comment by youkaichao** [comment]
Should modify ``_is_scalar ``.

```python
def _is_scalar(value: Any) -> bool:
    # check if the value is a scalar
    # 1. python bool object, number object: isinstance(value, Number)
    # 2. numpy scalar: isinstance(value, np.generic)
    # 3. python object rather than dict / Batch / tensor
    # the check of dict / Batch is omitted because this only checks a value.
    # a dict / Batch will eventually check their values
    if isinstance(value, (dict, Batch)):
        return False
    #=================== add these two lines
    if isinstance(value, torch.Tensor):
        return value.numel() == 1 and not value.shape
    value = np.asanyarray(value)
    return value.size == 1 and not value.shape

```

**Comment by duburcqa** [comment]
It could be added to https://github.com/thu-ml/tianshou/pull/147

**Comment by duburcqa** [comment]
Fixed by #147
